tests/Aspire.Hosting.Tests/PublishAsDockerfileTests.cs

```diff
@@ -198,10 +198,62 @@ public async Task PublishAsDockerFileConfigureContainer()
         Assert.Equal(expected, actual, ignoreLineEndingDifferences: true, ignoreWhiteSpaceDifferences: true);
     }
 
-    private static DisposableTempDirectory CreateDirectoryWithNodeDockerFile()
+    [Fact]
+    public async Task PublishProjectAsDockerFile()
+    {
+        var builder = TestDistributedApplicationBuilder.Create(DistributedApplicationOperation.Publish);
+
+        using var tempDir = CreateDirectoryWithDockerFile();
+
+        var path = tempDir.Directory.FullName;
+
+        var project = builder.AddProject("project", path, o => o.ExcludeLaunchProfile = true)
+                             .PublishAsDockerFile(c =>
+                             {
+                                 c.WithBuildArg("X", "y");
+                                 c.WithVolume("vol", "/app/shared");
+                             });
```

Comment:
Can you add args to the project resource just to ensure they don't get written to the manifest?